src/lmic/lmic_eu_like.c

```diff
@@ -103,7 +103,15 @@ ostime_t LMICeulike_nextJoinState(uint8_t nDefaultChannels) {
                 if (LMIC.datarate == LORAWAN_DR0)
                         failed = 1; // we have tried all DR - signal EV_JOIN_FAILED
                 else
+                {
+#if CFG_region != LMIC_REGION_as923
```

Comment:
This change is fine, but I will later stage a change to move this into the abstraction layer. (My long term goal is to allow one code-base to support several regions at once. That way a device could potentially adapt to its location if it moved across a border. For that to work, we have to minimize `#if` in the common code.)

Reply:
A single binary supporting several regions at once would be nice.  My interest is how you can tell the device what the region is changed ?

Reply:
There are two use cases: 

1. manual
2. automatic

For the manual case, I imagine shipping one binary to support multiple regions; the user manually sets the region upon installation (or manufacturing does this as the last step). This would have the advantage of avoiding conditional compiles, which means that things like CI go much faster (one build instead of N builds).  I consider the manual use case to be very helpful and good software engineering practice, even if automatic is impractical.

For the automatic case, the device might either have GPS/GNSS, or might look for class B network beacons in various bands, and decline to transmit until it has a solid idea of its location. Cellphones, for example, use for network beacons in order to figure out which bands to use. The device would have to use ADR-like techniques to keep track of whether it's still talking to the network. Once it decides that it's not, and decides to do a rejoin, the device has to determine location before transmitting.  

 A very careful device might also use an accelerometer and if moved "too far" might decide to drop back to "region hunting".

Reply:
For the automatic case, it would make a state machine complicate and it causes the size of image and memory usage might be bigger.  I would prefer the manual case.  But, it's surely wonderful feature.